packages/flutter/lib/src/rendering/animated_size.dart

```diff
@@ -68,6 +73,38 @@ class RenderAnimatedSize extends RenderAligningShiftedBox {
     _animation.curve = value;
   }
 
+  /// The [TickerProvider] for the [AnimationController] that runs the animation.
+  TickerProvider get tickerProvider => _tickerProvider;
+  TickerProvider _tickerProvider;
+  set tickerProvider(TickerProvider value) {
+    assert(value != null);
+    if (value == _tickerProvider)
+      return;
+    AnimationController oldController = _controller;
+    assert(oldController != null);
+    _setupAnimations(tickerProvider, _controller.duration, _animation.curve);
+    if (oldController.isAnimating) {
+      _controller.forward(from: oldController.value);
```

Comment:
I'm pretty sure that forward() will cause _this_ frame to have the value of the `from`, and the next frame will have advanced.

But I agree that it's cleaner to build the animation controller from the old one, so let me explore that.